add new `google_cloudbuild_trigger` resource

mathyourlife-fitbit opened the pull request:
Resource is not fully featured as the build triggers are still in beta, but allows for resources as the one used in the acceptance test.  All attributes marked as ForceNew as `.Patch()` calls have failed for me for when changing just something like the description.

```code
resource "google_cloudbuild_trigger" "build_trigger" {
  project  = "${google_project_services.acceptance.project}"
  description = "acceptance test build trigger"
  trigger_template {
    branch_name = "master"
    project     = "${google_project_services.acceptance.project}"
    repo_name   = "some-repo"
  }
  build {
    images = ["gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA"]
    tags = ["team a", "service b"]
    step {
      name = "gcr.io/cloud-builders/gsutil"
      args = "cp gs://mybucket/remotefile.zip localfile.zip "
    }
    step {
      name = "gcr.io/cloud-builders/go"
      args = "build my_package"
    }
    step {
      name = "gcr.io/cloud-builders/docker"
      args = "build -t gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA -f Dockerfile ."
    }
  }
}
```

mathyourlife-fitbit commented:
Will need to add a `website/docs/r/google_cloudbuild_trigger.html.markdown`

ndmckinley commented:
Are you ready for review?  Ping me when you are.  :)

kim0 commented:
Patiently waiting for this to be ready :)

mathyourlife-fitbit commented:
Sorry for the delay.

Documentation has been added and the branch rebased against master.  Travis shows an `The command "make website-test" exited with 2.` but test passes locally. Nav bar link and docs/providers/google/r/cloudbuild_trigger.html also render in local docker environment.

ndmckinley commented:
Great.  Working on that now.  The `make website-test` command is a little bit broken, so we can ignore that for now.  I am running the tests now.

ndmckinley commented on google/resource_cloudbuild_build_trigger.go:
```diff
@@ -0,0 +1,292 @@
+			build.Tags = convertStringArr(v.([]interface{}))
+		}
+		stepCount := d.Get("build.0.step.#").(int)
+		if stepCount > 0 {
```
I'd convert this to use make unconditionally, and then `append()` rather than `build.Steps[s] = step`; I'm not going to insist on it.  Either way, we have a pattern for these complex data structures, we like to have them in their own functions, parallel to the flatteners you already wrote (we call them expanders).

ndmckinley commented on google/resource_cloudbuild_build_trigger.go:
```diff
@@ -0,0 +1,292 @@
+		buildTrigger.Build = build
+	}
+
+	if d.Get("trigger_template.#").(int) > 0 {
```
Also its own function, please.  :)

ndmckinley commented:
And the tests passed, FYI!

mathyourlife-fitbit commented:
Thanks, @ndmckinley.  Added expanders to match flatteners and switched to make/cap for `[]*cloudbuild.BuildStep`.  Reran acceptance tests on my org, and passed again from this end.

mathyourlife-fitbit commented:
Thanks for the help, @ndmckinley 👍

naseemkullah commented:
Hello, will we be able to `terraform import` cloud build triggers as well?

Also can we create cloud build triggers for GitHub repos? My understanding is that it only works with GCP's Cloud Source Repos.. I'm looking for a way to use TF to create build triggers for private GitHub repos.

Many thanks in advance.
